pkg/controller/servicebindingrequest/sbrcontroller.go

```diff
@@ -113,18 +113,28 @@ func updateFunc(logger *log.Log) func(updateEvent event.UpdateEvent) bool {
 		isConfigMap := isOfKind(e.ObjectNew, "ConfigMap")
 
 		if isSecret || isConfigMap {
-			dataFieldsAreEqual, err := compareObjectFields(e.ObjectNew, e.ObjectOld, "data")
+			dataFieldsAreEqual, err := compareObjectFields(e.ObjectOld, e.ObjectNew, "data")
 			if err != nil {
 				logger.Error(err, "error comparing object fields: %s", err.Error())
 				return false
 			}
-
-			logger.Debug("Predicate evaluated", "dataFieldsAreEqual", dataFieldsAreEqual)
+			logger.Debug("Predicate evaluated for Secret/ConfigMap", "dataFieldsAreEqual", dataFieldsAreEqual)
 			return !dataFieldsAreEqual
 		}
+		specsAreEqual, err := compareObjectFields(e.ObjectOld, e.ObjectNew, "spec")
+		if err != nil {
+			logger.Error(err, "error comparing object's spec fields: %s", err.Error())
+			return false
```

Comment:
do we have unit test covering this case?

Reply:
yes